src/btree/bt_walk.c

```diff
@@ -657,13 +655,30 @@ int
 __wt_tree_walk_count(WT_SESSION_IMPL *session,
     WT_REF **refp, uint64_t *walkcntp, uint32_t flags)
 {
-	return (__tree_walk_internal(session, refp, walkcntp, NULL, flags));
+	return (__tree_walk_internal(
+	    session, refp, walkcntp, NULL, NULL, NULL, flags));
+}
+
+/*
+ * __wt_tree_walk_custom_skip --
+ *	Walk the tree calling a custom function to decide whether to skip refs.
+ */
+int
+__wt_tree_walk_custom_skip(
+    WT_SESSION_IMPL *session, WT_REF **refp,
+   int (*skip_func)(WT_SESSION_IMPL *, WT_REF *, void *, bool *),
+   void *func_context,
+   uint32_t flags)
+{
+	return (__tree_walk_internal(session, refp,
+	    NULL, NULL, skip_func, func_context, WT_READ_CUSTOM_SKIP | flags));
 }
 
 /*
  * __wt_tree_walk_skip --
  *	Move to the next/previous page in the tree, skipping a certain number
  *	of leaf pages before returning.
+ *	TODO: Fold this into the new custom skip function methodology.
```

Comment:
Because of where this gets done in the tree walk code, my guess is it's not worth the effort (but *shrug*).